### PR TITLE
Modify error message when atol=0 and rtol=0

### LINKED ISSUE #56377: Simplify error message for failed int comparison
If you use assertEqual to compare two integers you get a message like

```
AssertionError: False is not true : Scalars failed to compare as equal! Comparing -6 and 0 gives a difference of 6, but the allowed difference with rtol=0 and atol=0 is only 0!
```

The stuff about rtol and atol is is needlessly obscure and obscures the fact that, e.g., the integers in question are exit codes, for example. One easy fix is to suppress rtol/atol sentence if those are exactly zero; I'd also suggest we shouldn't report "difference" for integer comparison.

### PR DESCRIPTION
**Pull request by 1ntEgr8**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60898
* **#60897**

Fixes #56377
Example output: #60898

Differential Revision: [D29461107](https://our.internmc.facebook.com/intern/diff/D29461107)

### CODE REVIEW

File: torch/testing/_core.py [hunk 1]
```diff
@@ -224,11 +224,14 @@ def _helper(a, b, s) -> _compare_return_type:
 
         msg = None
         if not result:
-            msg = ("Comparing" + s + "{0} and {1} gives a "
-                   "difference of {2}, but the allowed difference "
-                   "with rtol={3} and atol={4} is "
-                   "only {5}!").format(a, b, diff,
-                                       rtol, atol, allowed_diff)
+            if rtol == 0 and atol == 0:
```
Comment (driazati) [inline torch/testing/_core.py:227]: given that we are in the future now, can you f-string-ify these?

**Review by driazati — APPROVED** [review]
Nice!

**Review by ezyang — APPROVED** [review]
GOD this has been bugging me forever. Much appreciated.

**Comment by 1ntEgr8** [comment]
@1ntEgr8 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29461107).

**Pull request closed without merging**